SwiftWisdomTests/StandardLibrary/String/String+IndexingTests.swift

```diff
@@ -2,6 +2,30 @@ import XCTest
 
 class String_IndexingTests: XCTestCase {
     
+    func testRangeSubscriptReturnsProperString() {
+        XCTAssertEqual("hello"[0..<0], "")
+        XCTAssertEqual("hello"[0..<1], "h")
+        XCTAssertEqual("hello"[0..<5], "hello")
```

Comment:
@bobgilmore Dennis and I talked about this a little. It seems like this function was originally written so that an out-of-bounds access would crash at runtime, much like array subscripting. I see the value in handling out-of-bounds in a better way, but didn't necessarily want to change the interface.

Some ideas are:
- Keep it as-is but add some alternative subscript(s) with safer behavior.
- Throw an error. This would break the existing interface.
- Change return type to `String?` and return `nil` when out-of-bounds. This also would break the existing interface.
- "Clamp" the given range to something in bounds and return the resultant string. This doesn't break the interface, but I'm not sure if that's the behavior most people would expect.